src/color.rs

```diff
@@ -779,6 +779,172 @@ impl Color {
+    /// 
+    /// // This will panic, as 1.1 is greater than the maximum accepted value of 1.0
+    /// let incorrect = original.saturate(1.1);
+    /// ```
+    /// 
+    /// Negative values will also result in a panic
+    /// 
+    /// ```should_panic
+    /// use turtle::Color;
+    /// 
+    /// let original = Color::hsl(120.0, 0.5, 0.9);
+    /// 
+    /// // This will panic, as a negative value is less than the minimum acceptable
+    /// // value of 0.0
+    /// let incorrect = original.saturate(-0.2);
+    /// ```
+    /// 
+    /// If you wish to desaturate a `Color` please see [`desaturate`].
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let original = Color::hsl(120.0, 0.3, 0.9);
+    /// 
+    /// // Instead of trying to saturate with a negative value as in the
+    /// // previous example, we simply desaturate by the same positive amount
+    /// let desaturated = original.desaturate(0.2);
+    /// assert_eq!(desaturated, Color::hsl(120.0, 0.1, 0.9));
+    /// ```
+    /// [Wikipedia Article]: https://en.wikipedia.org/wiki/HSL_and_HSV#Saturation
+    /// [`desaturate`]: ./struct.Color.html#method.desaturate
+    pub fn saturate(self, saturation: f64) -> Self {
+        assert_value_in_range!("saturation", saturation, 0., 1.);
+
+        let (h, s, l) = self.to_hsl();
+        let s_mod = if s + saturation <= 1. { s + saturation } else { 1. };
+        Color::hsla(h, s_mod, l, self.alpha) 
+    }
+
+    /// Create a new `Color` by decreasing the saturation level of this `Color` 
+    /// by the given percentage. The value is a float between
+    /// 0.0 and 1.0 indicating the percentage to increase the saturation.
+    /// So, if you wish to create a `Color` that is 30% more saturated than
+    /// this one, you would specify 0.3.
+    /// 
+    /// For more information on what saturation is in relation to HSL colors,
+    /// please see this [Wikipedia Article].
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let original = Color::hsl(120.0, 0.3, 0.9);
+    /// 
+    /// // Now, desaturate the original by 20%
+    /// let desaturated = original.desaturate(0.2);
+    /// assert_eq!(desaturated, Color::hsl(120.0, 0.1, 0.9));
+    /// ```
+    /// 
+    /// Since the minimum saturation value a color can have is 0%, attempting
+    /// to desaturate beyond that level will simply result in a `Color` with a
+    /// `saturation` value of 0.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let original = Color::hsl(120.0, 0.3, 0.9);
+    /// 
+    /// // Now, desaturate the color by 50%, which would equal -20%
+    /// let desaturated = original.desaturate(0.5);
+    /// 
+    /// // However the resulting color simply has a saturation of 0.0
+    /// assert_eq!(desaturated, Color::hsl(120.0, 0.0, 0.9));
+    /// ```
+    /// 
+    /// A color with a saturation level of 0.0 is known
+    /// as an [`achromatic`] color. As they have no hue, they are the range
```

Comment:
I really like that this sentence is here. :smile: 

The backticks should probably just be regular quotation marks.